newprinter.py

```diff
@@ -150,39 +150,15 @@ def download_gpg_fingerprint(url):
         debugprint('Not a https fingerprint URL: %s, ignoring driver' % url)
         return None
 
-    # Possible paths of a file with a set of SSL certificates which are
-    # considered trustworthy. The first one that exists will be used.
-    # This is used for downloading GPG key fingerprints for
-    # openprinting.org driver packages.
-    ssl_cert_file_paths = [
-        # Debian/Ubuntu use the ca-certificates package:
-        '/etc/ssl/certs/ca-certificates.crt',
-
-        # Fedora place the certificates in different locations:
-        '/etc/ssl/certs/ca-bundle.crt',
-        '/etc/pki/tls/certs/ca-bundle.crt'
-        ]
-
-    # default GPG key server
-    # this is the generally recommended DNS round-robin, but usually very
-    # slow:
-    #gpg_key_server = 'keys.gnupg.net'
-    gpg_key_server = 'hkp://keyserver.ubuntu.com:80'
-
-    cert = None
-    for f in ssl_cert_file_paths:
-        if os.path.exists(f):
-            cert = f
-            break;
-
-    if not cert:
-        debugprint('No system SSL certificates available for trust checking')
-        return None
-
```

Comment:
As I understand it, this is no longer necessary due to [PEP 476](https://www.python.org/dev/peps/pep-0476/). To test on an individual system:

`python3 -c "import urllib.request; urllib.request.urlopen('https://self-signed.badssl.com')"`

My understanding is that this code is present in Python 3.4 and all subsequent versions. So any distribution that is not EOL should have a Python 3 package that verifies certificates against the system certificate store by default.

Reply:
@slingamn thanks for the link, then it should be fine. However the code you've posted tracebacks because of self signed certificate in python3.9. The PEP mentions custom trust roots have to be used if self-signed certs should work.
I'm not entirely sure whether we just download from OpenPrinting, which has a correct certificate - it would be great if the 'except' branches for the request was enhanced to report if self-signed cert was in use - meaning to catch self-signed related exception and report it.

It would be a benefit for possible timeouts as well - catch the timeout related exception (I would guess the method will generate such once the timeout was hit) and report it - this way we can raise the timeout if the current causes problems.